File: Coding Sample/MS.NET/Fundamentals/Generic/Basics/genmethodtest1.cs
using System;

static class Test
{
	private static T Select<T>(int sign, T first, T second)
	{
		if(sign < 0)
			return first;
		
		return second;
	}

	public static void Main(string[] args)
	{
		int s = Convert.ToInt32(args[0]);

		string ss = Select(s, "Monday", "Tuesday");
		Console.WriteLine("Selected string = {0}", ss);

		double sd = Select(s, 8.25, 9.75);
		Console.WriteLine("Selected double = {0}", sd);

		long sl = Select(s, 123456L, 0xABCDEFL);
		Console.WriteLine("Selected long = {0}", sl);

	}
}

File: Coding Sample/Java/Fundamentals/Generic/Basics/GenMethodTest1.java
class GenMethodTest1{

	private static<T> T select(int sign, T first, T second){
		if(sign < 0)
			return first;
		return second;
	}
	
	public static void main(String[] args){
		int s = Integer.parseInt(args[0]);
		String ss = select(s, "Monday", "Tuesday");
		System.out.printf("Selected String = %s%n", ss);
		Interval si = select(s, new Interval(3, 45), new Interval(2, 10));
		System.out.printf("Selected Interval = %s%n", si);
		double sd = select(s, 4.56, 5.67);
		System.out.printf("Selected double = %s%n", sd);
		long sl = select(s, 123456L, 0xABCDEFL);
		System.out.printf("Selected long = %s%n", sl);
	}
}

